nomad/structs/config/vault.go

```diff
@@ -109,6 +111,9 @@ func (c *VaultConfig) AllowsUnauthenticated() bool {
 func (c *VaultConfig) Merge(b *VaultConfig) *VaultConfig {
 	result := *c
 
+	if b.Name != "" {
+		c.Name = b.Name
+	}
```

Comment:
Just out of curiosity, are there scenarios where we would merge `VaultConfig` with different names? Sounds like maybe a programmer error?

Reply:
There's probably a testing scenario where you have partial configs that you're merging. But that doesn't sound like a useful production code use case at all.

Reply:
Thinking about this a bit more, there are probably scenarios where you would want to create a new `VaultConfig` from the `DefaultVaultConfig`, so overwriting the name does make sense.